### PR TITLE
[FLINK-19152] Remove Kafka 0.10.x and 0.11.x connectors

### PR DESCRIPTION
**Pull request by aljoscha**
## What is the purpose of the change

Please see the ML thread for details: https://lists.apache.org/thread.html/r6d5c44e88d2d0fa187df6093694d6ad40c08561492c55b826b7231e7%40%3Cdev.flink.apache.org%3E

In short, 0.10.x and 0.11.x are very old and you can use the "modern" Kafka connector to connect to older brokers/clusters. Plus, if push comes to shove, users can use the code from 1.11 and compile it against a newer Flink release.

## Brief change log

- remove all Kafka 0.10 and 0.11 connector code, including sql connectors and tests
- update tests/examples to use modern connector where they were using the 0.10 connector
- update the documentation to remove mentions of 0.10 and 0.11

## Verifying this change

This doesn't add code, so no need for new tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: yes, we remove something
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

Documentation is updated to remove mentions of Kafka 0.10 and 0.11.

### CODE REVIEW

**Comment by aljoscha** [comment]
@sunjincheng121 Could you please take a look at the Python changes and maybe the changes to the Chinese documentation?

File: docs/dev/connectors/kafka.md [hunk 1]
```diff
@@ -48,34 +48,14 @@ For details on Kafka compatibility, please refer to the official [Kafka document
 </dependency>
```
Comment (dawidwys) [inline docs/dev/connectors/kafka.md:48]: Shall we update the start of the section as well?

```
Apache Flink ships with multiple Kafka connectors: universal, 0.10, and 0.11.
...
However, for Kafka versions 0.11.x and 0.10.x, we recommend using the dedicated ``0.11`` and ``0.10`` connectors, respectively.
```

File: docs/dev/connectors/kafka.md [hunk 2]
```diff
@@ -405,8 +385,7 @@ Consider setting appropriate [idelness timeouts]({{ site.baseurl }}/dev/event_ti
  
 ## Kafka Producer
```
Comment (dawidwys) [inline docs/dev/connectors/kafka.md:386]: Shall we update the `Migrating Kafka Connector from 0.11 to universal` section? I think the approach recommended there is not valid (at least for the producer).

File: docs/dev/connectors/kafka.md [hunk 3]
```diff
@@ -466,12 +445,12 @@ Through the producer record you can:
 ### Kafka Producers and Fault Tolerance
 
 <div class="codetabs" markdown="1">
-<div data-lang="Universal and 011" markdown="1">
-With Flink's checkpointing enabled, the `FlinkKafkaProducer011` (`FlinkKafkaProducer` for Kafka >= 1.0.0 versions) can provide
+<div data-lang="Universal" markdown="1">
```
Comment (dawidwys) [inline docs/dev/connectors/kafka.md:448]: How about we remove the Tabs whatsoever? IMO it does not make sense anymore for a single connector.

**Review by dawidwys — CHANGES_REQUESTED** [review]
I really like removing 13k+ lines! I think it is important to properly document the migration path. It does not need to be done in this PR, but let's at least create a blocker issue for 1.12  to add such a guide.

There are still dependencies on kafka-0.10/0.11 in:
* `pom.xml` in `flink-sql-client-test`
* flink-end-to-end-tests/README.md
* `org.apache.flink.streaming.kafka.test.base.KafkaEvent`
* `org.apache.flink.runtime.webmonitor.handlers.JarUploadHandlerTest`
* project-configuration.md (gradle project example)
* dev/table/connectors/kafka.md (especially in the description of `connector` option)
* nit: javadoc in `org.apache.flink.table.factories.Factory#factoryIdentifier` (we could change it to elasticsearch?)
* archetype-resources/pom.xml

File: docs/dev/connectors/kafka.md (hunk 2 above)
Comment (aljoscha) [reply docs/dev/connectors/kafka.md:386]: Ah yeah, I actually changed https://issues.apache.org/jira/browse/FLINK-19193 to make it work better for the Kafka connector. 👌

**Comment by aljoscha** [comment]
Thanks for the review! I pushed a commit that should address all comments.

File: docs/dev/connectors/kafka.md [hunk 4]
```diff
@@ -554,14 +551,26 @@ A mismatch in service name between client and server configuration will cause th
...
+
+The generic upgrade steps are outlined in [upgrading jobs and Flink versions
+guide]({{ site.baseurl }}/ops/upgrading.html). For Kafka, you additionally need
+to follow these steps:
+
+* Do not upgrade Flink and the Kafka Connector version at the same time.
+* Make sure you have a `group.id` configured for your Consumer.
+* Set `setCommitOffsetsOnCheckpoints(true)` on the consumer so that read
+  offsets are committed to Kafka. It's important to do this before stopping and
+  taking the savepoint. You might have to to a stop/restart cycle on the old
```
Comment (dawidwys) [inline docs/dev/connectors/kafka.md:564]:
```suggestion
  taking the savepoint. You might have to do a stop/restart cycle on the old
```

**Review by dawidwys — COMMENTED** [review]
I found some more references to old connectors:
* /dev/table/connectors/kafka.md: `With Flink's checkpointing enabled, the `kafka` and `kafka-0.11`...`
* some of the Chinese versions of the documentation (let's create a followup issue for that)
* there is also the `FlinkKafkaProducerMigrationOperatorTest`. Shall we remove the migration related code paths whatsoever? We can do that in a follow-up.

**Review by dawidwys — APPROVED** [review]

File: docs/dev/connectors/kafka.md (hunk 3 above)
Comment (aljoscha) [reply docs/dev/connectors/kafka.md:445]: I don't know how that happened but I thought I had already removed them. Good catch!

**Comment by aljoscha** [comment]
Thanks for the review! I would address the migration story as a follow-up. I know you already created a Jira issue for that.

**Pull request closed without merging**